lib/fbe/middleware/sqlite_store.rb

```diff
@@ -38,48 +35,34 @@ def delete(key)
 
   def write(key, value)
     value = JSON.dump(value)
-    perform do |tdb|
-      tdb.execute(<<~SQL, [key, value])
+    perform do |t|
+      t.execute(<<~SQL, [key, value])
         INSERT INTO cache(key, value) VALUES(?1, ?2)
         ON CONFLICT(key) DO UPDATE SET value = ?2
       SQL
     end
     nil
   end
 
-  def open
-    return if @db
-    @db = SQLite3::Database.new(@path)
-  end
-
-  def prepare
-    perform do |tdb|
-      tdb.execute 'CREATE TABLE IF NOT EXISTS cache(key TEXT UNIQUE NOT NULL, value TEXT);'
-      tdb.execute 'CREATE INDEX IF NOT EXISTS key_idx ON cache(key);'
-    end
-  end
-
-  def close
-    return if !@db || @db.closed?
-    @db.close
-    @db = nil
-  end
-
   def clear
     perform { _1.execute 'DELETE FROM cache;' }
   end
 
-  def drop
-    perform { _1.execute 'DROP TABLE IF EXISTS cache;' }
-  end
-
   def all
     perform { _1.execute('SELECT key, value FROM cache') }
   end
 
   private
 
   def perform(&)
+    @db ||=
+      SQLite3::Database.new(@path).tap do |d|
+        d.transaction do |t|
+          t.execute 'CREATE TABLE IF NOT EXISTS cache(key TEXT UNIQUE NOT NULL, value TEXT);'
+          t.execute 'CREATE INDEX IF NOT EXISTS key_idx ON cache(key);'
+        end
+        at_exit { @db.close if @db && !@db.closed? }
```

Comment:
@Yegorov I can't imagine a situation when `@db` is nil or it's already closed. How can this happen?

Reply:
@yegor256 the situation is extremely rare, but it can happen if after 64 line process receive signal `SIGTERM` (in this case `@db` not initialized yet). 
Сheck `!@db.closed?` really unnecessary.
Do you agree to leave only `if @db`?